lib/src/test/java/com/thenewboston/utils/BankApiMockEngine.kt

```diff
@@ -1,79 +1,137 @@
+                        }
+                        else -> {
+                            error("Unhandled ${request.url.encodedPath}")
                         }
                     }
-                    BankAPIJsonMapper.SINGLE_VALIDATOR_ENDPOINT -> {
-                        val content = BankAPIJsonMapper.mapValidatorToJson()
-                        if (enableErrorResponse) {
-                            respond(errorContent, HttpStatusCode.InternalServerError, responseHeaders)
-                        } else {
-                            respond(content, HttpStatusCode.OK, responseHeaders)
+                }
+            }
+
+            installJsonFeature()
+        }
+
+    private fun MockRequestHandleScope.sendResponse(
+        content: String,
+        errorContent: String,
+        isError: Boolean
+    ) = when {
+        isError -> respond(errorContent, InternalServerError, responseHeaders)
+        else -> respond(content, HttpStatusCode.OK, responseHeaders)
+    }
+
+    private fun patchBankEngine(
+        enableErrorResponse: Boolean = false,
+        isInvalidResponse: Boolean = false
+    ) = HttpClient(MockEngine) {
+        val errorContent = BankAPIJsonMapper.mapInternalServerErrorToJson()
+
+        engine {
+            addHandler { request ->
+                when {
+                    request.url.encodedPath == BankAPIJsonMapper.BANKS_TRUST_ENDPOINT -> {
+                        val content = BankAPIJsonMapper.mapBankTrustResponseToJson()
+                        val invalidContent = BankAPIJsonMapper.mapInvalidBankTrustResponseToJson()
+                        when {
+                            enableErrorResponse -> respond(
+                                errorContent,
+                                InternalServerError,
+                                responseHeaders
+                            )
+                            isInvalidResponse -> respond(
+                                invalidContent,
+                                Accepted,
+                                responseHeaders
+                            )
+                            else -> respond(content, Accepted, responseHeaders)
                         }
                     }
-                    BankAPIJsonMapper.CONFIG_ENDPOINT -> {
-                        val content = BankAPIJsonMapper.mapBankDetailToJson()
-                        if (enableErrorResponse) {
-                            respond(errorContent, HttpStatusCode.InternalServerError, responseHeaders)
-                        } else {
-                            respond(content, HttpStatusCode.OK, responseHeaders)
+                    request.url.encodedPath.startsWith(BankAPIJsonMapper.ACCOUNTS_ENDPOINT) -> {
+                        val requestBodyString = (request.body as TextContent).text
+                        val requestedTrust =
+                            Json.decodeFromString<UpdateTrustRequest>(requestBodyString).message.trust
```

Comment:
Is this not a bit redundant? Would the mapper class not map this trust double to a string for us?

Reply:
I want to the mock engine to read the trust from the request and respond with the corresponding trust. Didn't find an easier way to get the trust from the request but I'm sure there's easier ways.